addon/instance-initializers/hot-loader-livereload-plugin.js

```diff
@@ -7,26 +7,31 @@ function createPlugin (appName, hotReloadService) {
   Plugin.identifier = 'ember-hot-reload';
   Plugin.version = '1.0'; // Just following the example, this might not be even used
   Plugin.prototype.reload = function(path) {
-        // TODO: is this needed? Consider removing since it might have undersired effects
-    window.runningTests = true;
-    var tags = document.getElementsByTagName('script');
-    for (var i = tags.length; i >= 0; i--){
-      if (tags[i] && tags[i].getAttribute('src') != null && tags[i].getAttribute('src').indexOf(appName) !== -1) {
-        tags[i].parentNode.removeChild(tags[i]);
+    const cancelableEvent = { modulePath: path, cancel: false};
+    hotReloadService.trigger('willLiveReload', cancelableEvent);
+    if (cancelableEvent.cancel) {   // Only hotreload if someone canceled the regular reload
+      // TODO: is this needed? Consider removing since it might have undersired effects
+      window.runningTests = true;
+      var tags = document.getElementsByTagName('script');
+      for (var i = tags.length; i >= 0; i--){
+        if (tags[i] && tags[i].getAttribute('src') != null && tags[i].getAttribute('src').indexOf(appName) !== -1) {
+          tags[i].parentNode.removeChild(tags[i]);
+        }
       }
-    }
-    var script = document.createElement('script');
-    script.onload = function() {
-      setTimeout(function() {
-        window.runningTests = false;
-        hotReloadService.trigger('newChanges', path);
-      }, 10);
-    };
-    script.type = 'text/javascript';
-    script.src = `/assets/${appName}.js`;
-    document.body.appendChild(script);
+      var script = document.createElement('script');
+      script.onload = function() {
+        setTimeout(function() {
+          window.runningTests = false;
+          hotReloadService.trigger('newChanges', path);
```

Comment:
It might just be the event names what I don't like. Any suggestions? Maybe `willLiveReload` and `willHotReload` or `willLiveReload` and `liveReloadAborted`...